### PR TITLE
Chore: Bundle - Replace Deprecated Output Escaping

### LINKED ISSUE #37101: [Issue] Chore: Bundle - Replace Block Escaping with Escaper
This issue is automatically created based on existing pull request: magento/magento2#37088: Chore: Bundle - Replace Block Escaping with Escaper

---------
### Description (*)
Refactors the `Magento_Bundle` module to replace `$block` escaping functions with `$escaper` escaping functions.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by pykettk**
### Description (*)
Refactors the `Magento_Bundle` module to replace `$block` escaping functions with `$escaper` escaping functions.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#37101: Chore: Bundle - Replace Block Escaping with Escaper

### CODE REVIEW

File: app/code/Magento/Bundle/view/adminhtml/templates/catalog/product/edit/tab/attributes/extend.phtml [hunk 1]
```diff
@@ -3,35 +3,33 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+declare(strict_types=1);
```
Comment (Den4ik) [inline app/code/Magento/Bundle/view/adminhtml/templates/catalog/product/edit/tab/attributes/extend.phtml:6]: I believe that no make sense use it in phtml files.

File: app/code/Magento/Bundle/view/adminhtml/templates/product/composite/fieldset/options/type/checkbox.phtml [hunk 2]
```diff
@@ -3,16 +3,24 @@
...
  * See COPYING.txt for license details.
  */
+declare(strict_types=1);
+
+use Magento\Bundle\Block\Adminhtml\Catalog\Product\Composite\Fieldset\Options\Type\Checkbox;
+use Magento\Framework\Escaper;
+
+/** @var Escaper $escaper */
+/** @var Checkbox $block */
+$_option = $block->getOption();
```
Comment (Den4ik) [inline app/code/Magento/Bundle/view/adminhtml/templates/product/composite/fieldset/options/type/checkbox.phtml:13]: Due to file update, could you please update variable names and remove underscore?

**Review by Den4ik — CHANGES_REQUESTED** [review]
@pykettk, thanks for work on this PR.
Please look at my comments

File: app/code/Magento/Bundle/view/adminhtml/templates/catalog/product/edit/tab/attributes/extend.phtml (hunk 1 above)
Comment (pykettk) [reply app/code/Magento/Bundle/view/adminhtml/templates/catalog/product/edit/tab/attributes/extend.phtml:6]: Why not?

File: app/code/Magento/Bundle/view/adminhtml/templates/product/composite/fieldset/options/type/checkbox.phtml (hunk 2 above)
Comment (pykettk) [reply app/code/Magento/Bundle/view/adminhtml/templates/product/composite/fieldset/options/type/checkbox.phtml:13]: Is this a blocking request for change? I'd rather not refactor more than I have to. While I agree that removing leading underscores from variable names is preferred, not doing it as part of this change doesn't make anything worse than it already is.

File: app/code/Magento/Bundle/view/adminhtml/templates/catalog/product/edit/tab/attributes/extend.phtml (hunk 1 above)
Comment (Den4ik) [reply app/code/Magento/Bundle/view/adminhtml/templates/catalog/product/edit/tab/attributes/extend.phtml:6]: We don't use any type declarations in phtml files. phtml files are representation and should not contain any logic and should be refactored if you find such of them

Comment (pykettk) [reply app/code/Magento/Bundle/view/adminhtml/templates/catalog/product/edit/tab/attributes/extend.phtml:6]: Are you going to block this PR if these changes are not made?

File: app/code/Magento/Bundle/view/adminhtml/templates/product/composite/fieldset/options/type/checkbox.phtml (hunk 2 above)
Comment (pykettk) [reply app/code/Magento/Bundle/view/adminhtml/templates/product/composite/fieldset/options/type/checkbox.phtml:13]: Are you going to block this PR if these changes are not made?

File: app/code/Magento/Bundle/view/adminhtml/templates/catalog/product/edit/tab/attributes/extend.phtml (hunk 1 above)
Comment (Den4ik) [reply app/code/Magento/Bundle/view/adminhtml/templates/catalog/product/edit/tab/attributes/extend.phtml:6]: @pykettk it's not blocking pr, but in my opinion it doesn't match standards.

**Comment by Den4ik** [comment]
@magento run all tests

File: app/code/Magento/Bundle/view/adminhtml/templates/sales/order/view/items/renderer.phtml [hunk 3]
```diff
@@ -3,30 +3,28 @@
...
-$catalogHelper = $block->getData('catalogHelper');
-?>
+/** @var Escaper $escaper */
+/** @var SecureHtmlRenderer $secureRenderer */
+/** @var Renderer $block */
+$_item = $block->getItem();
+$items = $block->getChildren($_item);
+$_count = count($items);
+$_index = 0;
+$canEditItemQty = true;
```
Comment (Den4ik) [inline app/code/Magento/Bundle/view/adminhtml/templates/sales/order/view/items/renderer.phtml:20]: Looks like this variable become from another file

File: app/code/Magento/Bundle/view/adminhtml/templates/sales/order/view/items/renderer.phtml [hunk 4]
```diff
@@ -3,30 +3,28 @@
...
+/** @var Data $catalogHelper */
+$catalogHelper = $block->getData('catalogHelper');
 
-<?php if ($block->getOrderOptions() || $_item->getDescription() || $block->canDisplayGiftmessage()): ?>
-    <?php $_showlastRow = true ?>
-<?php else: ?>
-    <?php $_showlastRow = false ?>
-<?php endif; ?>
+$_prevOptionId = '';
+$_showlastRow = $block->getOrderOptions() || $_item->getDescription();
```
Comment (Den4ik) [inline app/code/Magento/Bundle/view/adminhtml/templates/sales/order/view/items/renderer.phtml:26]: Missed `$block->canDisplayGiftmessage()` check

File: app/code/Magento/Bundle/view/adminhtml/templates/sales/order/view/items/renderer.phtml [hunk 5]
```diff
@@ -3,30 +3,28 @@
...
-<?php $_index = 0 ?>
-<?php
-/** @var \Magento\Catalog\Helper\Data $catalogHelper */
-$catalogHelper = $block->getData('catalogHelper');
-?>
+/** @var Escaper $escaper */
+/** @var SecureHtmlRenderer $secureRenderer */
+/** @var Renderer $block */
+$_item = $block->getItem();
+$items = $block->getChildren($_item);
```
Comment (Den4ik) [inline app/code/Magento/Bundle/view/adminhtml/templates/sales/order/view/items/renderer.phtml:17]:
```suggestion
$items = array_merge([$_item], $_item->getChildrenItems())
```

File: app/code/Magento/Bundle/view/adminhtml/templates/sales/invoice/view/items/renderer.phtml [hunk 6]
```diff
@@ -3,30 +3,28 @@
...
-$catalogHelper = $block->getData('catalogHelper');
-?>
+/** @var Escaper $escaper */
+/** @var SecureHtmlRenderer $secureRenderer */
+/** @var Renderer $block */
+$_item = $block->getItem();
+$items = $block->getChildren($_item);
+$_count = count($items);
+$_index = 0;
+$canEditItemQty = true;
```
Comment (Den4ik) [inline app/code/Magento/Bundle/view/adminhtml/templates/sales/invoice/view/items/renderer.phtml:20]: Looks like this variable become from another file

File: app/code/Magento/Bundle/view/adminhtml/templates/sales/shipment/create/items/renderer.phtml [hunk 7]
```diff
@@ -3,27 +3,28 @@
...
-<?php $_count = count($items) ?>
-<?php $_index = 0 ?>
+/** @var Escaper $escaper */
+/** @var SecureHtmlRenderer $secureRenderer */
+/** @var Renderer $block */
+$_item = $block->getItem();
+$items = $block->getChildren($_item);
+$_count = count($items);
+$_index = 0;
+$canEditItemQty = true;
```
Comment (Den4ik) [inline app/code/Magento/Bundle/view/adminhtml/templates/sales/shipment/create/items/renderer.phtml:20]: This variable is redundant

File: app/code/Magento/Bundle/view/adminhtml/templates/sales/shipment/view/items/renderer.phtml [hunk 8]
```diff
@@ -3,28 +3,28 @@
...
-$catalogHelper = $block->getData('catalogHelper');
-?>
+/** @var Escaper $escaper */
+/** @var SecureHtmlRenderer $secureRenderer */
+/** @var Renderer $block */
+$_item = $block->getItem();
+$items = $block->getChildren($_item);
+$_count = count($items);
+$_index = 0;
+$canEditItemQty = true;
```
Comment (Den4ik) [inline app/code/Magento/Bundle/view/adminhtml/templates/sales/shipment/view/items/renderer.phtml:20]: This variable is redundant

File: app/code/Magento/Bundle/view/adminhtml/templates/sales/shipment/view/items/renderer.phtml [hunk 9]
```diff
@@ -3,28 +3,28 @@
...
-<?php $_index = 0 ?>
-<?php
-/** @var \Magento\Catalog\Helper\Data $catalogHelper */
-$catalogHelper = $block->getData('catalogHelper');
-?>
+/** @var Escaper $escaper */
+/** @var SecureHtmlRenderer $secureRenderer */
+/** @var Renderer $block */
+$_item = $block->getItem();
+$items = $block->getChildren($_item);
```
Comment (Den4ik) [inline app/code/Magento/Bundle/view/adminhtml/templates/sales/shipment/view/items/renderer.phtml:17]:
```suggestion
$items = array_merge([$_item->getOrderItem()], $_item->getOrderItem()->getChildrenItems());
$shipItems = $block->getChildren($_item);
```

File: app/code/Magento/Bundle/view/frontend/templates/sales/order/items/renderer.phtml [hunk 10]
```diff
@@ -3,21 +3,26 @@
...
+/** @var Escaper $escaper */
+/** @var LocaleFormatter $localeFormatter */
+/** @var Renderer $block */
 $parentItem = $block->getItem();
-$viewModel = $block->getViewModel();
-$items = $viewModel->getOrderItems((int)$parentItem->getOrderId(), (int)$parentItem->getId());
-$index = 0;
-$prevOptionId = '';
+$items = $block->getChildren($parentItem);
+$_order = $block->getItem()->getOrderItem()->getOrder();
```
Comment (Den4ik) [inline app/code/Magento/Bundle/view/frontend/templates/sales/order/items/renderer.phtml:17]:
```suggestion
$viewModel = $block->getViewModel();
$items = $viewModel->getOrderItems((int)$parentItem->getOrderId(), (int)$parentItem->getId());
```

File: app/code/Magento/Bundle/view/frontend/templates/sales/order/shipment/items/renderer.phtml [hunk 11]
```diff
@@ -3,26 +3,30 @@
...
+use Magento\Bundle\Block\Sales\Order\Items\Renderer;
+use Magento\Framework\Escaper;
+use Magento\Framework\Locale\LocaleFormatter;
+
+/** @var Escaper $escaper */
+/** @var LocaleFormatter $localeFormatter */
+/** @var Renderer $block */
+$parentItem = $block->getItem();
+$items = $block->getChildren($parentItem);
+$_order = $block->getItem()->getOrderItem()->getOrder();
```
Comment (Den4ik) [inline app/code/Magento/Bundle/view/frontend/templates/sales/order/shipment/items/renderer.phtml:17]:
```suggestion
$items = array_merge([$parentItem->getOrderItem()], $parentItem->getOrderItem()->getChildrenItems());
$shipItems = $block->getChildren($parentItem);
```

**Review by Den4ik — CHANGES_REQUESTED** [review]
@pykettk Added new update requests

**Pull request closed without merging**